ring/util.go

```
@@ -102,27 +102,28 @@ func WaitRingStability(ctx context.Context, r *Ring, op Operation, minStability,
 
 	// Get the initial ring state.
 	ringLastState, _ := r.GetAllHealthy(op) // nolint:errcheck
-	ringLastStateTs := time.Now()
 
-	const pollingFrequency = time.Second
-	pollingTicker := time.NewTicker(pollingFrequency)
-	defer pollingTicker.Stop()
+	// This timer will be our state
+	stabilityTimer := time.NewTimer(minStability)
+
+	stateWatch, _ := r.statesWatcher.Register()
```

Why ignore the error?

---

fair